### PR TITLE
Fanmodes

### PR DESCRIPTION
**Pull request by pastaq**

### CODE REVIEW

File: ayn-sensors.c [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * control via hwmon sysfs, as well as temperature sensor readings 
  * exposed by the EC and RGB control via platform sysfs.
  *
- * Fan control is provided via pwm interface in the range [0-255].
+ * Fan control is provided via pwm interface in the range [0-254].
  * They use [0-128] as range in the EC, the written value is
```
Comment (BoukeHaarsma23) [inline ayn-sensors.c:8]: Is this not 0-127?

**Review by BoukeHaarsma23 — COMMENTED** [review]
I could add some few nits if you'd like. But LGTM

File: ayn-sensors.c (hunk 1 above)
Comment (pastaq) [reply ayn-sensors.c:8]: Userspace tools expect 0-254. The EC accepts 0-127. We do a bit shift to normalize the userspace input to the EC input.

Comment (pastaq) [reply ayn-sensors.c:8]: Oh, you meant the comment... nvm

Comment (pastaq) [reply ayn-sensors.c:8]: [EC-Design-Check (translated).xlsx](https://github.com/ShadowBlip/ayn-platform/files/12382229/EC-Design-Check.translated.xlsx)
Per the docs from Ayn its 128, but it is possible that this doc is wrong as well.

**Pull request merged**